common/transport/mqtt/src/mqtt_base.ts

```diff
@@ -257,3 +256,13 @@ export class MqttBase extends EventEmitter {
     this._mqttClient.end(forceDisconnect, callback);
   }
 }
+
+export namespace MqttBase {
+  export interface TransportConfig {
+    host: string;
+    sharedAccessSignature?: string | SharedAccessSignature;
+    deviceId: string;
+    x509?: X509;
+  }
+}
```

Comment:
Do you really need to create a new type namespace and interface.  I was able to dispense with this in the provisioning code.

Reply:
Yeah, I'm not crazy about this, but I wanted to reduce the number of changes in the actual implementation. 

---
In reply to: [142805734](https://github.com/Azure/azure-iot-sdk-node/pull/101#discussion_r142805734) [](ancestors = 142805734)